builtin/credential/userpass/path_login.go

```diff
@@ -86,14 +86,22 @@ func (b *backend) pathLogin(ctx context.Context, req *logical.Request, d *framew
 	// Check for a password match. Check for a hash collision for Vault 0.2+,
 	// but handle the older legacy passwords with a constant time comparison.
 	passwordBytes := []byte(password)
-	if !legacyPassword {
+	switch {
```

Comment:
Do you prefer the switch-case to:

```
if legacyPassword && subtle.ConstantTimeCompare(...) {
  logic
  return 
} else {
 if err := bcrypt...
}
...
```

One advantage of the `if/else` like above is we could reduce nesting, but I defer to your preference.

Reply:
Works for me!